### ISSUE TITLE
previous_install.yaml logic breaks when rke2 is installed with an RPM and then later switched to TAR

### ISSUE THREAD

**Issue by mddamato** [rancherfederal/rke2-ansible#193]

**Comment by jcox10** [comment]
It also breaks when you install RKE2 with the RPM, and then use Rancher to upgrade RKE2. You end up with the installed RPM being one version, but the running RKE2 version being different.

**Comment by aleiner** [comment]
Switching between RPMs and Tarballs as the means to install RKE2 isn't something we want to support at the moment.

@jcox10, can you elaborate on what you said? DId you use rke2-ansible to build a downstream cluster and import it to Rancher?

**Comment by jcox10** [comment]
@aleiner Correct, built cluster using rke2-ansible then imported to Rancher. Rancher then expects to manage all of the updates using the [System Update Controller](https://github.com/rancher/system-upgrade-controller). If RKE2 is updated via RPM then the Rancher version gets out of sync. Also allowing the RPM to upgrade is a bit dangerous as it can occur when you are not expecting it such as when doing `dnf update`.  I talked to RGS support a bit about this, and they said SUSE doesn't really support doing RPM upgrades on a Rancher managed cluster.

So what I have is a bit of ansible to just set the RPM repo `enabled=0` after the cluster is built. Then nothing changes when running updates, and Rancher MCM is happy.

**Comment by jcox10** [comment]
Yes and the `rke2_installed` variable from `previous_install.yml` can be used as the flag, to detect if RKE2 is already present on the system. But you'll need to have those tasks run for the RPM install by commenting out the where clause for `install_method == tarball`

**Comment by aleiner** [comment]
Addressed via #326 . Currently in the 2.1.0 branch, but will be moved to main shortly.
